liberty-maven-plugin/src/main/java/net/wasdev/wlp/maven/plugins/server/PluginConfigSupport.java

```diff
@@ -191,4 +193,29 @@ protected String stripVersionFromName(String name, String version) {
             return name;
         }
     }
+    
+    protected void addAppConfiguration(String artifactId) throws Exception {
+   
+        log.warn(messages.getString("info.install.app.not.configured"));
+         
+        // Add webApplication configuration into the target server.xml. 
+        File serverXML = new File(serverDirectory, "server.xml");
+        ServerXmlDocument.addAppElment(serverXML, artifactId);
+        
+        log.info(MessageFormat.format(messages.getString("info.install.app.add.configuration"), artifactId));
+    }
+    
+    protected boolean isApplicationConfigured() throws Exception {
```

Comment:
Also, you need to parse both server.xml and server include xml as described in the WAD:
~~~
Add webApplication configuration into the target server.xml.  Produce  a warning
message that the application  is not configured in the source server.xml  or
included config.
~~~